src/PositionManager/PositionManager.cpp

```diff
@@ -70,6 +71,8 @@ void QGCPositionManager::_positionUpdated(const QGeoPositionInfo &update)
     }
     if (newGCSPosition != _gcsPosition) {
         _gcsPosition = newGCSPosition;
+        _gcsHeading = update.attribute(QGeoPositionInfo::Direction);
```

Comment:
What if the heading changed from the previous update, but the position did not? This code will not run to update the heading.

Reply:
Gps signals don't usually stay really fixed, but who knows what data the users might feed in here?
Fixed it.